### PR TITLE
Remove kubescheduler.config.k8s.io/v1alpha1

### LINKED ISSUE #89260: Remove kubescheduler.config.k8s.io/v1alpha1
/sig scheduling
/kind cleanup

As we get ready to move to beta, we should first get rid of the v1alpha1 API to reduce maintenance cost.

In particular, this would free us from having to maintain Algorithm Source.

Before I open this for contributors, I would like to know if there are any concerns.

### PR DESCRIPTION
**Pull request by gavinfish**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Get rid of the scheduler config v1alpha1 API to reduce maintenance cost.

**Which issue(s) this PR fixes**:

Fixes #89260

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Remove kubescheduler.config.k8s.io/v1alpha1
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

kubernetes/enhancements#785

### CODE REVIEW

File: cmd/kube-scheduler/app/options/options_test.go [hunk 1]
```diff
@@ -134,31 +134,6 @@ leaderElection:
 		t.Fatal(err)
 	}
 
-	v1alpha1Config := filepath.Join(tmpDir, "kubeconfig_v1alpha1.yaml")
```
Comment (alculquicondor) [inline cmd/kube-scheduler/app/options/options_test.go]: Could you add unit tests for `Deprecated.AlgorithmProvider` and `Deprecated.PolicyConfigFile`?

Comment (alculquicondor) [reply cmd/kube-scheduler/app/options/options_test.go]: Actually, not here, I'll leave a comment somewhere else

**Comment by alculquicondor** [comment]
/approve
for scheduling

/assign @lavalamp

File: cmd/kube-scheduler/app/options/options_test.go (hunk 1 above)
Comment (gavinfish) [reply cmd/kube-scheduler/app/options/options_test.go]: Sure, I will follow up if the origin assignee does not continue the work.

**Comment by alculquicondor** [comment]
ping @lavalamp

**Pull request merged**